src/uharfbuzz/_harfbuzz.pyx

```diff
@@ -1057,3 +1057,146 @@ cdef class DrawFuncs:
+            l[i].objidx = links[i][2]
+
+class RepackerError(Exception):
+    pass
+
+def repack(subtables: List[bytes],
+           graphnodes: List[Tuple[List[Tuple[int, int, int]],
+                                  List[Tuple[int, int, int]]
+                                  ]]) -> bytes:
+
+    """The whole table is represented as a Graph
+       and the input graphnodes is a flat list of subtables
+       with each node(subtable) represented by a tuple of 2
+       list: real_link list and virtual_link list.
+
+       A link(egde) is an offset link between parent table and
+       child table. It's represented in the format of a tuple: 
+       (posiiton: int, width: int, objidx: int):
+
+       - position: means relative position of the offset field
+         in bytes from the beginning of the subtable's C struct.
+         e.g:
+         a GSUB header struct in C looks like below:
+         
+         uint16 majorVersion
+         uint16 minorVersion
+         offset16 scriptListOffset
+         offset16 featureListOffset
+         offset16 lookupListOffset
+         And the position for scriptListOffset is 4 
+         which is calculated from (16+16)/8
+       - width: size of the offset:
+         e.g 2 for offset16 and 4 for offset32
+       - objidx: objidx is the index of the subtable
+         in graph/tree generated by postorder traversal
+
+       There're 2 types of links:
+       - real_link represents an real offset field in the parent table
+       - virtual_link is not real offset link, it specifies an ordering
+         constraint that harfbuzz packing must follow.
+         A virtual link would have 0 width, 0 position, and a real objidx.
+         e.g:
+         if Node A has a virtual link with objidx b(corresponding Node is B),
+         then that means Node B is always packed after Node A in the final
+         serialized order
+         """
+
+    if len(subtables) != len(graphnodes):
+        raise ValueError(
+            f"Input num of subtables({len(subtables)}) != num of graph nodes({len(graphnodes)})"
+        )
+    cdef:
+        unsigned int num_nodes = len(subtables)
+        bytes table_bytes = b''.join(subtables)
+        char* table_data = table_bytes
+        unsigned int head = 0, tail = 0
+        hb_link_t* p = NULL
+    try:
+        obj_list = HBObject(num_nodes)
+        for i in range(num_nodes):
+            tail += len(subtables[i])
+            obj_list.update_obj_length(i, table_data + head, table_data + tail)
+            head = tail
+            
+            node = graphnodes[i]
+            #real_links
+            p = obj_list.create_links(i, len(node[0]), 1)
+            if p != NULL:
+                obj_list.update_links(i, 1, node[0])
+            #virtual_links
+            p = obj_list.create_links(i, len(node[1]), 0)
+            if p != NULL:
+                obj_list.update_links(i, 0, node[1])
+    except MemoryError:
+        raise MemoryError()
```

Comment:
This is re-raising the same exception, which makes the try except block redundant and can be done without.